src/servers/src/prom.rs

```diff
@@ -699,3 +741,57 @@ fn promql_expr_to_metric_name(expr: &PromqlExpr) -> Option<String> {
         }
     }
 }
+
+#[axum_macros::debug_handler]
+pub async fn series_query(
+    State(handler): State<PromHandlerRef>,
+    Query(params): Query<LabelsQuery>,
+    Form(form_params): Form<LabelsQuery>,
+) -> Json<PromJsonResponse> {
+    let mut queries: Vec<String> = params.matches.0;
+    if queries.is_empty() {
+        queries = form_params.matches.0;
+    }
+    if queries.is_empty() {
+        return PromJsonResponse::error("Unsupported", "match[] parameter is required");
+    }
+
+    let start = params
+        .start
+        .or(form_params.start)
+        .unwrap_or_else(yesterday_rfc3339);
+    let end = params
+        .end
+        .or(form_params.end)
+        .unwrap_or_else(current_time_rfc3339);
+
+    let db = &params.db.unwrap_or(DEFAULT_SCHEMA_NAME.to_string());
+    let (catalog, schema) = super::parse_catalog_and_schema_from_client_database_name(db);
+    let query_ctx = Arc::new(QueryContext::with(catalog, schema));
+
+    let mut series = Vec::new();
+    for query in queries {
+        let prom_query = PromQuery {
+            query,
+            start: start.clone(),
+            end: end.clone(),
+            // TODO: find a better value for step
+            step: DEFAULT_LOOKBACK_STRING.to_string(),
+        };
+
+        let result = handler.do_query(&prom_query, query_ctx.clone()).await;
+
+        let response = retrieve_series_from_query_result(result, &mut series).await;
+
+        if let Err(err) = response {
+            // Prometheus won't report error if querying nonexist label and metric
```

Comment:
Let's return the errors right now, we will support a compatible mode in future.

Reply:
Do you mean:

```rust
retrieve_series_from_query_result(result, &mut series).await?;
```

Reply:
Yes, always return the error even when table is not found etc.